### PR TITLE
add typescript support

### PR DESCRIPTION
**Pull request by lhz516**
Hi, I'd like to do some contributions!

- Add typescript support to this project
- Update Webpack and Babel configs
- Add missing eslint rules so that the existing code won't have errors

### CODE REVIEW

**Comment by justinmc** [comment]
@lhz516 Thanks for the contribution and sorry for the late response!  I just updated a bunch of dependencies, which caused conflicts.  Would you be able to pull master and fix the conflicts?  Let me know if anything is confusing.  I'd love to get typescript support in.

**Comment by lhz516** [comment]
Cool. I've just based and resolved the conflict.

I also added `karma` related dev deps for resolving an npm audit issue. Now there's 0 audit issue.

You can generate the type file by `npm run build:d.ts`

File: package.json [hunk 1]
```diff
@@ -43,12 +48,18 @@
     "eslint-plugin-jsx-a11y": "^6.2.3",
     "eslint-plugin-react": "^7.19.0",
     "file-loader": "^3.0.1",
+    "jasmine": "^3.4.0",
     "jest": "^25.4.0",
+    "karma": "^5.0.2",
+    "karma-chrome-launcher": "^2.2.0",
+    "karma-jasmine": "^2.0.1",
+    "karma-webpack": "^3.0.5",
```
Comment (justinmc) [inline package.json:56]: Are you sure you need the karma and jasmine dependencies?  I just swapped them out for jest instead.  Nothing should be using karma anymore, unless I forgot something.

Comment (lhz516) [reply package.json:56]: Ah, I missed that. Please remove them if no more vulnerabilities with `npm audit`.

**Comment by lhz516** [comment]
I just removed them and everything looks fine.

**Comment by justinmc** [comment]
Looks good, thanks again for doing this!  I'll release a new version 0.13.0 right now.

**Comment by justinmc** [comment]
Done.

**Pull request merged**